test/tools/provisioner-cli/main.go

```diff
@@ -43,16 +43,22 @@ func main() {
 		installDirectory = "../../install"
 	}
 
+	action := flag.String("action", "provision", "string")
+	flag.Parse()
+
+	pv.Action = *action
+
 	cfg := envconf.New()
 
+	if *action == "uploadimage" {
+		os.Setenv("TEST_PROVISION", "yes")
```

Comment:
What's the reason for setting TEST_PROVISION to "yes" for uploadImage. It feels like `TEST_PROVISION` should be linked to cluster creation rather than image actions? Is it because of some other tech debt restriction from elsewhere? If so, and it can't be updated, can you add this to the commit message body please, or as a comment in the code. Thanks!

Reply:
I suggested this addition. The reason I had in mind is if `TEST_PROVISION` is not set to `yes` or `true` then in `provision_ibmcloud_initializer.go` we check the existing VPC infrastructure parameters like `SSH_KEY_ID` and `VPC_SUBNET_ID` are set, but not `REGION` and `APIKEY` which you need to upload a new Custom Image.

If you want to be able to use the `uploadimage` action without setting `TEST_PROVISION` to `yes`, then for properties that are only checked when `TEST_PROVISION` is not set to `yes` you need, if you're doing `uploadimage`, to check `REGION` and `APIKEY` are set.

I don't suggest adding those whole paragraphs in as a code comment but we need to find an elegant way to work around the mix of the action parameter for the provisioner CLI and the environment variables.

Or maybe we solve the above by doing a bigger piece of refactoring out the use of environment variables in the provisoner CLI `main` package and the `provisioner` package where the properties are checked.

Reply:
I've chatted to Tia about whether, instead of setting `TEST_PROVISION` in the main, we can update the `REGION` and `APIKEY` validation checks in `provision_ibmcloud_initializer.go` to run if `TEST_PROVISION=yes`, *or* `action == "uploadimage"`. I think this change to reduce the reliance on `TEST_PROVISION` is probably better and keeps the validation logic together.